modules/schema/src/main/java/org/apache/ignite/internal/schema/DecimalNativeType.java

```diff
@@ -57,13 +57,6 @@ public int scale() {
         return scale;
     }
 
-    /** {@inheritDoc} */
-    @Override public boolean mismatch(NativeType type) {
-        return super.mismatch(type)
-            || precision < ((NumericNativeType)type).precision
-            || scale < ((NumericNativeType)type).scale;
```

Comment:
Can we validate Decimal somehow?
E.g. check if `precision - scale >= type.precision - type.scale ` ?

I mean we always can convert types with different scales (truncate or expand), 
but must ensure if precision will be compatible after the conversion.

Reply:
If we validate here, we will discard even those values that we could have stored. 
For example, there is a special case with zero value string representations:

System.out.println(new BigDecimal("0.00").scale()); //2
System.out.println(new BigDecimal("0E-2").scale()); //2
System.out.println(new BigDecimal("0E+2").scale()); //-2

All of these expressions are "zero" and we can definitely store them. But we cannot understand this here.